internal/repos/github_test.go

```diff
@@ -124,6 +124,63 @@ func TestGitHub_stripDateRange(t *testing.T) {
 	}
 }
 
+func TestPublicRepos_PaginationTerminatesGracefully(t *testing.T) {
+	setUpRcache(t)
+
+	fixtureName := "GITHUB-ENTERPRISE/list-public-repos"
+	gheToken := prepareGheToken(t, fixtureName)
+
+	service := &types.ExternalService{
+		Kind: extsvc.KindGitHub,
+		Config: extsvc.NewUnencryptedConfig(marshalJSON(t, &schema.GitHubConnection{
+			Url:   "https://ghe.sgdev.org",
+			Token: gheToken,
+		})),
+	}
+
+	factory, save := newClientFactory(t, fixtureName)
+	defer save(t)
+
+	ctx := context.Background()
+	githubSrc, err := NewGitHubSource(ctx, logtest.Scoped(t), service, factory)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	results := make(chan *githubResult)
+	go func() {
+		githubSrc.listPublic(context.Background(), results)
+		close(results)
+	}()
+
+	count := 0
+	countArchived := 0
+	for result := range results {
+		if result.err != nil {
+			t.Errorf("unexpected error: %s, expected repository instead", result.err.Error())
+		}
+		if result.repo.IsArchived {
+			countArchived++
+		}
+		count++
+	}
+	if count != 100 {
+		t.Errorf("unexpected repo count, wanted: 100, but got: %d", count)
+	}
+	if countArchived != 1 {
+		t.Errorf("unexpected archived repo count, wanted: 1, but got: %d", countArchived)
+	}
+}
+
+func prepareGheToken(t *testing.T, fixtureName string) string {
+	gheToken := os.Getenv("GHE_TOKEN")
```

Comment:
probably we can add:
```suggestion
func prepareGheToken(t *testing.T, fixtureName string) string {
    t.Helper()
	gheToken := os.Getenv("GHE_TOKEN")
```